### PR TITLE
WC: Persistance update + a bunch of visual bugfixes

### LINKED ISSUE #20357: [User story] Show connected dApps
### User Story

As a user, I want to see my connected dApps, so that I know my current connections and can disconnect dApps.

[User story and acceptance criteria](https://www.notion.so/I-want-to-see-my-connected-dApps-ac2e612c163d447ba969d53c365750a7?pvs=4)

### LINKED ISSUE #20358: [User story] Connect to dApps
# Feature Issue

[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to connect to a dApp via WalletConnect, so that I can interact with the dApp though my wallet

[User story and acceptance criteria](https://www.notion.so/I-want-to-connect-to-a-dApp-via-WalletConnect-0efa85076296435ba79f7d1893f048f1?pvs=4)

### LINKED ISSUE #20753: Wallet Connect: can't connect to certain DApps
# Bug Report

## Problem 

When scanning the QR code for some DApps, an error appears in the Status app:

Screenshot 2024-07-15 at 16 18 03

DApps from the [list](https://www.notion.so/Testing-Wallet-Connect-interactions-0739f176856848999f0ea1671e5dfbe0) that cannot be connected to:

https://aave.com/
https://compound.finance/
https://rarible.com/

#### Expected behavior
Can connect to all DApps from the [list](https://www.notion.so/Testing-Wallet-Connect-interactions-0739f176856848999f0ea1671e5dfbe0)

#### Actual behavior

https://github.com/user-attachments/assets/89c6addc-bf71-4286-a147-d4dd6f7f06d3

### Reproduction

1. Open any of these DApps: 
https://aave.com/
https://compound.finance/
https://rarible.com/
2. `Connect Wallet` > Select `Wallet Connect`
3. Scan the QR code with the Status app scanner

### Additional Information

- Status version: nightly 15/07
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by alwx**
Fixes #20357 
Fixes [#20358](https://github.com/status-im/status-mobile/issues/20358)
Fixes #20782
Fixes https://github.com/status-im/status-mobile/issues/20753

A bunch of bugfixes I made when going through these two user stories:
- https://www.notion.so/I-want-to-connect-to-a-dApp-via-WalletConnect-0efa85076296435ba79f7d1893f048f1?pvs=4
- https://www.notion.so/I-want-to-see-my-connected-dApps-ac2e612c163d447ba969d53c365750a7

The most important change is the usage of db everywhere — it was the only big missing piece that was not implemented properly since the screens were implemented much earlier than the db was implemented.
Also it includes some visual changes, and also some bigger stuff:
1) Corrected check for unsupported networks (previously it was not showing the incorrected network toast when non-eip155 were specified in a request)
2) Support for selecting the correct account (the first one when scanning with global QR code scanner & the selected one when scanning with the QR scanner for a particular wallet)

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/wallet_connect/core.cljs [hunk 1]
```diff
@@ -85,9 +85,11 @@
 
 (defn required-networks-supported?
   [proposal supported-networks]
-  (let [required-networks (get-in proposal [:params :requiredNamespaces :eip155 :chains])
-        supported-eip155  (set (map chain-id->eip155 supported-networks))]
-    (every? #(contains? supported-eip155 %) required-networks)))
+  (let [required-namespaces (get-in proposal [:params :requiredNamespaces])]
+    (when (= (keys required-namespaces) '(:eip155))
```
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/core.cljs]: 👍

File: src/quo/components/drawers/bottom_actions/style.cljs [hunk 2]
```diff
@@ -21,12 +21,13 @@
                        (colors/theme-colors colors/white colors/neutral-95 theme))})
 
 (defn buttons-container
-  [actions]
-  {:flex-direction     (if (= actions :two-vertical-actions) :column :row)
-   :justify-content    :space-around
-   :padding-vertical   12
-   :gap                12
-   :padding-horizontal 20})
+  [actions buttons-container-style]
```
Comment (J-Son89) [inline src/quo/components/drawers/bottom_actions/style.cljs]: nit: can just be `container-style` here

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 3]
```diff
@@ -54,7 +54,8 @@
  (fn [{:keys [db]} [proposal]]
    (log/info "Received Wallet Connect session proposal: " {:id (:id proposal)})
    (let [accounts                     (get-in db [:wallet :accounts])
-         without-watched              (remove :watch-only? (vals accounts))
+         current-viewing-address      (get-in db [:wallet :current-viewing-account-address])
+         availabled-accounts          (remove :watch-only? (vals accounts))
```
Comment (shivekkhurana) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs]: `availabled-accounts` seem to be a typo, did you want `available-accounts` ?

**Review by shivekkhurana — APPROVED** [review]

File: src/quo/components/drawers/bottom_actions/style.cljs (hunk 2 above)
Comment (shivekkhurana) [reply src/quo/components/drawers/bottom_actions/style.cljs]: I also feel `container-style` is more apt.

**Review by clauxx — COMMENTED** [review]
Thanks for the PR!

**Review by clauxx — APPROVED** [review]

File: src/quo/components/drawers/bottom_actions/style.cljs (hunk 2 above)
Comment (alwx) [reply src/quo/components/drawers/bottom_actions/style.cljs:24]: it cannot be since there is already `container-style`

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 4]
```diff
@@ -54,7 +54,8 @@
  (fn [{:keys [db]} [proposal]]
    (log/info "Received Wallet Connect session proposal: " {:id (:id proposal)})
    (let [accounts                     (get-in db [:wallet :accounts])
-         without-watched              (remove :watch-only? (vals accounts))
+         current-viewing-address      (get-in db [:wallet :current-viewing-account-address])
+         available-accounts           (remove :watch-only? (vals accounts))
```
Comment (smohamedjavid) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs]: We need to filter the operable accounts (the `operable?` boolean key in the account would be helpful) in addition to removing the watched accounts. The user won't be able to sign any request if the key pair of that account is missing/not imported. I checked the `personal_sign` method from a non-operable account and it failed.

Additionally, we need to remove it from the account selection in the connection request sheet.

File: src/status_im/contexts/wallet/connected_dapps/view.cljs [hunk 5]
```diff
@@ -83,7 +83,8 @@
   []
   (let [{:keys [bottom]}                   (safe-area/get-insets)
         {:keys [color] :as wallet-account} (rf/sub [:wallet/current-viewing-account])
-        sessions                           (rf/sub [:wallet-connect/persisted-sessions])
+        sessions                           (rf/sub
+                                            [:wallet-connect/persisted-sessions-for-current-account])
```
Comment (clauxx) [inline src/status_im/contexts/wallet/connected_dapps/view.cljs]: we shouldn't rely on the persisted sessions by default, but rather on the ones coming from web3wallet, as they will be most up-to-date. We should just use the persisted ones in cases when the user has no internet connection and keep it synced with the state from wallet-connect active sessions.

File: src/status_im/contexts/wallet/wallet_connect/effects.cljs [hunk 6]
```diff
@@ -56,13 +49,6 @@
        (promesa/then on-success)
        (promesa/catch on-fail))))
 
-(rf/reg-fx
- :effects.wallet-connect/fetch-active-sessions
- (fn [{:keys [web3-wallet on-success on-fail]}]
-   (-> (wallet-connect/get-active-sessions web3-wallet)
-       (promesa/then on-success)
-       (promesa/catch on-fail))))
-
```
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/effects.cljs:62]: we should use the active sessions as the source of truth for "connected dapps" i think

File: src/status_im/contexts/wallet/connected_dapps/view.cljs (hunk 5 above)
Comment (clauxx) [reply src/status_im/contexts/wallet/connected_dapps/view.cljs]: cc: @stefandunca we are approaching it the same way in desktop, right?

File: src/status_im/subs/root.cljs [hunk 7]
```diff
@@ -169,7 +169,7 @@
 (reg-root-key-sub :wallet-connect/web3-wallet :wallet-connect/web3-wallet)
 (reg-root-key-sub :wallet-connect/current-proposal :wallet-connect/current-proposal)
 (reg-root-key-sub :wallet-connect/current-request :wallet-connect/current-request)
-(reg-root-key-sub :wallet-connect/pairings :wallet-connect/pairings)
+(reg-root-key-sub :wallet-connect/persisted-sessions :wallet-connect/persisted-sessions)
```
Comment (clauxx) [inline src/status_im/subs/root.cljs]: should be called "active-sessions" IMO. Don't think we should differentiate between the persisted ones and the ones we get from web3-wallet cause the persisted ones are just a fallback

File: src/status_im/contexts/wallet/wallet_connect/core.cljs [hunk 8]
```diff
@@ -85,12 +85,19 @@
...
+            supported-eip155  (set (map chain-id->eip155 supported-networks))]
+        (every? #(contains? supported-eip155 %) required-networks)))))
 
 (defn get-networks-by-mode
   [db]
   (let [test-mode? (get-in db [:profile/profile :test-networks-enabled?])
         networks   (get-in db [:wallet :networks (if test-mode? :test :prod)])]
     (mapv #(-> % :chain-id) networks)))
+
+(defn event-should-be-handled?
```
Comment (shivekkhurana) [inline src/status_im/contexts/wallet/wallet_connect/core.cljs:103]: This function also fixes an issue raised here: https://github.com/status-im/status-mobile/pull/20693#issuecomment-2228288723

TL;DR; If a user connected an app on `testnet` and then switched to `mainnet`. A WC request sent on `testnet` will show up and fail. 

But it needs one more clause, if the event is not meant to be handled, it should be rejected. So WC can send new events in future. 

This is the apt error response: 
```  USER_REJECTED_CHAINS: {
    message: "User rejected chains.",
    code: 5001,
  },
  ```
  
  In this function you can do something like: 
  
  ```clj
(defn event-should-be-handled? []
  (let [to-handle? (some ...)]
    (when-not to-handle?
      (rf/dispatch [:respond-with-rejection] ()))
    to-handle?)) 
```

Here's an example on how to send a response: 
```clj
(rf/reg-event-fx
 :wallet-connect/reject-session-request
 (fn [_ _]
   {:fx [[:dispatch
          [:wallet-connect/send-response
           {:error (wallet-connect/get-sdk-error
                    constants/wallet-connect-user-rejected-error-key)}]]]}))
```

The constant in this case will be `USER_REJECTED_CHAINS`

**Review by shivekkhurana — APPROVED** [review]
The PR is fine as it is, but I added a comment that will fix a peripheral issue. Lmk if you want to do it in your PR. If its too much, I'll branch out on top of your work.

File: src/status_im/subs/root.cljs (hunk 7 above)
Comment (shivekkhurana) [reply src/status_im/subs/root.cljs]: `persisted-sessions` might be different from active sessions as the user can disconnect from the dapp when status-app is closed. 

I introduced this term `persisted-sessions` in the UI when I wrote the handlers for saving and retrieving sessions.

Comment (alwx) [reply src/status_im/subs/root.cljs]: I've replaced it with just `sessions`, because the list consists of persisted sessions first and then get replaced with active sessions once they are loaded from the SDK (which means it would be inaccurate to always call it `active sessions`)

File: src/status_im/contexts/wallet/wallet_connect/core.cljs (hunk 8 above)
Comment (alwx) [reply src/status_im/contexts/wallet/wallet_connect/core.cljs:103]: @shivekkhurana it's probably worth creating a separate PR, otherwise this one will get postponed for some time again :) so feel free to raise it

File: src/status_im/constants.cljs [hunk 9]
```diff
@@ -282,6 +282,7 @@
 (def ^:const wallet-connect-supported-events #{"accountsChanged" "chainChanged"})
 (def ^:const wallet-connect-session-proposal-event "session_proposal")
 (def ^:const wallet-connect-session-request-event "session_request")
+(def ^:const wallet-connect-session-delete-event "session_delete")
```
Comment (clauxx) [inline src/status_im/constants.cljs:290]: why not the `disconnect` event from the [docs](https://docs.walletconnect.com/walletkit/react-native/usage#evm-methods--events). Or the docs again out of sync with implementation 😆?

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 10]
```diff
@@ -204,10 +198,59 @@
...
+   (let [sessions' (mapv (fn [{:keys [sessionJson] :as session}]
+                           (assoc session
+                                  :accounts
+                                  (-> sessionJson
+                                      types/json->clj
+                                      :namespaces
+                                      :eip155
+                                      :accounts)))
+                         sessions)]
+     {:fx [[:dispatch [:wallet-connect/fetch-active-sessions]]]
```
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs:240]: Maybe we should put all this logic of figuring out the sessions based on the active and persisted ones inside one effect (or better a function used by the effect). It could be in its own namespace i.e. `sessions`. Otherwise it might become difficult to manage & read with all the different events and "success/failed" events, which call other effects and so on. It's not necessary to change it now as it might be late for that with the deadline and all, but IMO would be really good to move all this logic outside the events eventually and try to keep the logic in functions instead of directly inside events/effects that call each other. What do you think?

File: src/status_im/subs/root.cljs (hunk 7 above)
Comment (clauxx) [reply src/status_im/subs/root.cljs]: @shivekkhurana indeed they're different, but what we're storing in the db is not exactly the persisted sessions, but rather the sessions that are active. It's more of an implementation detail that we're falling back on the persisted-sessions when there's no internet connection.

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 11]
```diff
@@ -204,10 +198,59 @@
...
+
+(rf/reg-event-fx
+ :wallet-connect/fetch-active-sessions
+ (fn [{:keys [db]}]
+   (let [web3-wallet (get db :wallet-connect/web3-wallet)]
+     {:fx [[:effects.wallet-connect/fetch-active-sessions
+            {:web3-wallet web3-wallet
+             :on-fail     #(log/error "Failed to get active sessions" {:error %})
+             :on-success  #(rf/dispatch [:wallet-connect/fetch-active-sessions-success
+                                         (js->clj % :keywordize-keys true)])}]]})))
```
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs]: would be better if the conversion happens in the effect, as I don't think we'll ever need the js object.

File: src/status_im/contexts/wallet/wallet_connect/session_proposal/view.cljs [hunk 12]
```diff
@@ -60,13 +65,11 @@
   (let [accounts         (rf/sub [:wallet/accounts-without-watched-accounts])
```
Comment (smohamedjavid) [inline src/status_im/contexts/wallet/wallet_connect/session_proposal/view.cljs]: Please use the `wallet/operable-accounts-without-watched-accounts` sub to display only the operable accounts.

File: src/status_im/subs/wallet/wallet_connect.cljs [hunk 13]
```diff
@@ -37,14 +38,24 @@
...
+     (->> sessions
+          (filter (fn [session]
+                    (= dapp-url (get session :url))))
           (first)))))
 
+(rf/reg-sub
+ :wallet-connect/sessions-for-current-account
+ :<- [:wallet-connect/sessions]
+ :<- [:wallet/current-viewing-account]
+ (fn [[sessions {:keys [address]}]]
```
Comment (smohamedjavid) [inline src/status_im/subs/wallet/wallet_connect.cljs]: The `wallet/current-viewing-account-address` sub would be best here as we need only the address. 

The `wallet/current-viewing-account` might trigger unwanted re-renders if the account is updated from the signal.

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 14]
```diff
@@ -104,28 +117,18 @@
...
      {:fx [[:effects.wallet-connect/disconnect
             {:web3-wallet web3-wallet
-             :topic       topic
+             :topic       pairing-topic
              :on-fail     on-fail
-             :on-success  on-success}]]})))
+             :on-success  (fn []
+                            (rf/dispatch [:wallet-connect/disconnect-session pairing-topic])
+                            (when on-success
+                              (on-success)))}]]})))
```
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs:129]: same thing here about having a single effect/event pair for handling the disconnection and avoid the unnecessary on-success -> dispatch etc. It could also be in the `sessions` ns.

**Review by clauxx — APPROVED** [review]

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 15]
```diff
@@ -204,10 +198,59 @@
...
+;; We first load sessions from database, then we initiate a call to Wallet Connect SDK and
+;; then replace the list we have stored in the database with the one that came from the SDK.
+;; In addition to that, we also update the backend state by marking sessions that are not
+;; active anymore by calling `:wallet-connect/disconnect-session`.
+(rf/reg-event-fx
+ :wallet-connect/fetch-active-sessions-success
+ (fn [{:keys [db]} [sessions]]
+   (let [time-now           (-> (time/now)
+                                time.coerce/to-long
+                                (/ 1000))
```
Comment (smohamedjavid) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs]: Suggestion: The `now` cofx in the event serves the timestamp and we can utilize it here.

**Comment by pavloburykh** [comment]
@alwx thanks for the PR. Could you please ping me up once all review comments will be resolved? Thanks

File: src/status_im/constants.cljs (hunk 9 above)
Comment (alwx) [reply src/status_im/constants.cljs:290]: https://docs.walletconnect.com/walletkit/web/usage#session-disconnect

According to this (and to reality as well), `session_delete` is getting fired when the disconnection by another peer happens. The docs are a bit of a garbage but you can check it the following way:
- connect a dapp (react app works fine)
- click "disconnect" in react app itself
- observe the `session_delete` event being received and handled properly

File: src/status_im/contexts/wallet/wallet_connect/events.cljs (hunk 10 above)
Comment (alwx) [reply src/status_im/contexts/wallet/wallet_connect/events.cljs:240]: Let me create an issue and handle it separately if you don't mind. Gonna be a bit of a nice refactoring but maybe better to handle it after we finish with the core functionality.

**Comment by clauxx** [comment]
@pavloburykh I believe all comments were addressed. Should be good to test 👍

**Comment by pavloburykh** [comment]
@alwx thanks for the PR. Please take a look at the issues. 

### ISSUE 1 Unable to connect to Dapps: scanning QR results in errors. 

**Steps:** 

1. Enable Mainnet in Status app
2. Select mainnet in Dapp (find the list of the dApps in Actual result below)
3. Generate Wallet connect QR on Dapp side
4. Scan QR via universal scanner in Status app
5. Observe the result

**Actual result:** Connection modal does not appear. Scanning results in `Oops. This QR does't work with Status` or `app requires an unsupported network` errors. 

1. https://app.uniswap.org/swap?chain=mainnet (`Uniswap requires an unsupported network`)
2. https://app.aave.com/?marketName=proto_mainnet (`Oops. This QR does't work with Status`)
3. https://stake.lido.fi/ (`Stake with Lido | Lido requires an unsupported network`)
4. https://app.compound.finance/?market=weth-mainnet (`Oops. This QR does't work with Status`)
5. https://dydx.trade/DYDX (`dydx requires an unsupported network`)
6. https://zapper.xyz/dashboard?modal=connect (`zapper requires an unsupported network`)
7. https://rarible.com/connect (`Oops. This QR does't work with Status`)
8. https://opensea.io/ (`opensea requires an unsupported network`)

At the same time @qoqobolo was able to connect to some of the above dApps, but also facing same issues with others. 

[Status-debug-logs - 2024-07-17T125238.894.zip](https://github.com/user-attachments/files/16263633/Status-debug-logs.-.2024-07-17T125238.894.zip)

photo_2024-07-17 12 42 40
photo_2024-07-17 12 43 53

**Comment by qoqobolo** [comment]
NOTE to ISSUE 1: there is a separate issue https://github.com/status-im/status-mobile/issues/20753 for the following dapps:
https://app.aave.com/
https://app.compound.finance/
https://rarible.com/

All three dapps have a different error - `Oops. This QR does't work with Status`, so it might be a separate issue.
cc @alwx @pavloburykh

**Comment by alwx** [comment]
@qoqobolo @pavloburykh that's fixed now and ready to be re-checked

**Comment by clauxx** [comment]
> NOTE to ISSUE 1: there is a separate issue #20753 for the following dapps: https://app.aave.com/ https://app.compound.finance/ https://rarible.com/
> 
> All three dapps have a different error - `Oops. This QR does't work with Status`, so it might be a separate issue. cc @alwx @pavloburykh

@qoqobolo Made a PR which fixes these [here](https://github.com/status-im/status-mobile/pull/20782). The others should be already fixed here.

**Comment by pavloburykh** [comment]
### ISSUE 2 `Undefined is not a function` when declining connection request

**Steps:** 

1. Scan wallet connect QR
2. Tap decline in connection modal
3. Observe the result

**Actual result:** `Undefined is not a function` error appears. 

[Status-debug-logs - 2024-07-18T113910.949.zip](https://github.com/user-attachments/files/16280210/Status-debug-logs.-.2024-07-18T113910.949.zip)

https://github.com/user-attachments/assets/79da100e-0889-4c80-bd27-94f96bf7fb86


**Expected result:** no error

**Comment by pavloburykh** [comment]
### ISSUE 3 Disconnected Dapp is removed from the list only after app relogin

**Steps:** 

1. Connect to any dAapp
2. Disconnect dAapp on dApp's side
3. See if dApp disappeared from dApps list in Status app

**Actual result:** dApp dissappears from the list only after relogin

https://github.com/user-attachments/assets/d7e8c769-8635-4681-b3d6-a2956c4f9be9

**Expected result:** dApp should disappear from the list without need to relogin

**Comment by clauxx** [comment]
> ### ISSUE 3 Disconnected Dapp is removed from the list only after app relogin
> **Steps:**
> 
> 1. Connect to any dAapp
> 2. Disconnect dAapp on dApp's side
> 3. See if dApp disappeared from dApps list in Status app
> 
> **Actual result:** dApp dissappears from the list only after relogin
> 
>  telegram-cloud-document-2-5242528426862089751.1.mp4 
> **Expected result:** dApp should disappear from the list without need to relogin

@alwx could it be that some dApps send the `disconnect` event instead of `session_delete`?

**Comment by pavloburykh** [comment]
### ISSUE 4 Disconnection of dApp in Status app does not result in disconnection on dApp's side


**Steps:** 

1. Connect to any dApp
2. Disconnect dApp in Status app
3. See if wallet is disconnected on dApp's side 

**Actual result:** wallet is still displayed as connected on dApp's side despite it has been disconnected in Status app

**Expected result:** wallet should be disconnected on dApp's side

**Comment by clauxx** [comment]
@alwx @qoqobolo @pavloburykh I think would be good to merge this PR sooner than later and make issues out of the bugs (except the most severe ones that should be fixed here). The branch is already becoming big and fixing some of the  bugs depends on it. I think the ISSUE 2 (decline) and 1 must be fixed here, but the disconnect ones could be included in a single issue. WDYT?

**Comment by pavloburykh** [comment]
> @alwx @qoqobolo @pavloburykh I think would be good to merge this PR sooner than later and make issues out of the bugs (except the most severe ones that should be fixed here). I think the ISSUE 2 (decline) and 1 must be fixed here, but the disconnect ones could be created as a single issue for all 3 bugs. WDYT?

Sure @alwx. Need some more time to finish testing.

**Comment by pavloburykh** [comment]
### ISSUE 5 Dapp requests are displayed for wrong Users restored within one Status app

**Steps:** 

1. Create/restore User A and User B within one Status app
2. Login via User A
3. Connect to dApp
4. Logout User A and login via User B
5. Trigger any dApp request for User A
6. Observe the result

**Actual result:** request modal appears while being logged as User B


https://github.com/user-attachments/assets/617e40e6-9c1e-46ec-bc05-23abba79849e



**Expected result:** request modal should not appear as it has been triggered for another user

**Comment by pavloburykh** [comment]
### ISSUE 6 Last account that established connection is not selected when connecting through global scanner

**Preconditions:** User has Account 1 and Account 2

**Steps:** 

1. Connect Account 2 to any dApp
2. Scan wallet connect QR of another dApp using global scanner
3. See which account is selected for connection

**Expected result:** Account 2 should be selected. According to the [requirements](https://www.notion.so/I-want-to-connect-to-a-dApp-via-WalletConnect-0efa85076296435ba79f7d1893f048f1) : 

(9+) I want to connect to a dApp via WalletConnect 2024-07-18 13-26-19

**Actual result:** Account 1 is selected


https://github.com/user-attachments/assets/6b9dd988-a667-4646-9fdd-5fcf673bed1c

**Comment by pavloburykh** [comment]
### ISSUE 7 No error is shown when connecting to dApp without internet connection

**Steps:** 

1. Enable airplane mode on mobile device
2. Scan wallet connect QR and tap connect
3. See if any error is shown informing that connection has not been established

**Expected result:** we should inform user that connection has not been established because network is unavailable

**Actual result:** no error


https://github.com/user-attachments/assets/872bcd40-4696-46e1-bdcf-577338e006c9

**Comment by pavloburykh** [comment]
### ISSUE 8 Connecting to some dApps using wrong network does not result in error on Status app side

For example let's take https://rarible.com/connect

**Steps:** 

1. Enable testmode in Status app
2. https://rarible.com/connect remains on Mainnet
3. Generate wallet connect QR
4. Scan QR - tap connect

**Expected result:** `app requires unsupported network` error should be shown when scanning QR

**Actual result:** no errors are shown on Status app side, user can proceed with connection. On dApps side connection will result in errors. 

See videos below from Status app and https://rarible.com/connect

**Status app:** 

https://github.com/user-attachments/assets/c0be935d-824b-422b-8201-a1522c876877

**Rarible:** 

https://github.com/user-attachments/assets/f0f1d9ee-efeb-459c-803e-54a3666f75b1

**Comment by pavloburykh** [comment]
@alwx thanks for the PR. ISSUE 1 is fixed. Rest of the issues will be logged separately once this PR is merged.

**Comment by pavloburykh** [comment]
@alwx @clauxx  created separate issues
https://github.com/status-im/status-mobile/issues/20796
https://github.com/status-im/status-mobile/issues/20797
https://github.com/status-im/status-mobile/issues/20798
https://github.com/status-im/status-mobile/issues/20800
https://github.com/status-im/status-mobile/issues/20801
https://github.com/status-im/status-mobile/issues/20802
https://github.com/status-im/status-mobile/issues/20803
